docs/contracts.rst

```diff
@@ -695,7 +695,7 @@ Taking the following contract code as an example:
     >>> array_contract.functions.getBytes2Value().call()
     [b'b\x00']
     >>> array_contract.functions.setBytes2Value([b'a']).transact()
-    HexBytes('0x39bc9a0bf5b8ec8e8115ccb20bf02f5570351a20a8fd774da91353f38535bec1')
+    HexBytes('0x977bd66e0ef59433a02c2f71f61ec3ca48bf7a5a19d10aece84570061fe9d4f3')
```

Comment:
Hm, I don't follow, why did this change?

Reply:
Good question. I didn't think too hard about it changing. Would the different VM rules (istanbul vs. constantinople) do it?

Reply:
Hm, the only thing I can think of is that maybe the estimated gas usage changed? I would definitely want to make sure that's the only thing that changed before merging, though. Maybe by getting the transaction in the old version and new version. Could even add that to the doc test, like
```py
>>> transaction_hash = array_contract.functions.setBytes2Value([b'a']).transact()
>>> w3.eth.getTransaction(transaction_hash)
{
  value: 0,
  gas_price: 1000000000,
  // etc
}
```
Then it should be really obvious if the gas price is the only thing that changes. We could potentially skip the gas cost in the test with `...` to be a bit more flexible without losing much.

Reply:
So one option is to add that ^ doctest change as a separate PR without the eth-tester upgrade. Then rebase this PR on top, to see how it changes cleanly.

(or if you're feeling a little more adventurous with local rebasing, you could push it as a separate commit in this PR, but need to push it alone to trigger CI and let it finish, before adding the eth-tester upgrade)

Reply:
Yep, looks like it was gas/gasPrice changing in the tx_params. I printed out the result from `w3.eth.getTransaction` and took the higher gas and gasPrice and rounded up then added it to the transaction params. Let me know if you'd like me to do something different/see something else!